buildSrc/src/integTest/groovy/org/opensearch/gradle/internal/InternalDistributionArchiveSetupPluginFuncTest.groovy

```diff
@@ -119,53 +118,64 @@ class InternalDistributionArchiveSetupPluginFuncTest extends AbstractGradleFuncT
                 consumeArchive project(path: ':producer-tar', configuration:'default' )
             }
-
             tasks.register("copyDir", Copy) {
                 from(configurations.consumeDir)
                 into('build/dir')
             }
-
             tasks.register("copyArchive", Copy) {
                 from(configurations.consumeArchive)
                 into('build/archives')
             }
         }
         """
-        when:
-        def result = gradleRunner("copyArchive").build()
-
-        then: "tar task executed and target folder contains plain tar"
-        result.task(':buildProducerTar').outcome == TaskOutcome.SUCCESS
-        result.task(':consumer:copyArchive').outcome == TaskOutcome.SUCCESS
-        file("producer-tar/build/distributions/opensearch-min.tar.gz").exists()
-        file("consumer/build/archives/opensearch-min.tar.gz").exists()
 
-        when:
-        result = gradleRunner("copyDir", "-Pversion=1.0").build()
-        then: "plain copy task executed and target folder contains plain content"
-        result.task(':buildProducer').outcome == TaskOutcome.SUCCESS
-        result.task(':consumer:copyDir').outcome == TaskOutcome.SUCCESS
-        file("producer-tar/build/install/someFile.txt").exists()
-        file("producer-tar/build/install/snapshot-1.0.txt").exists()
-        file("consumer/build/dir/someFile.txt").exists()
+        if (OperatingSystem.current() != OperatingSystem.WINDOWS) {
+            when:
+            def result = gradleRunner("copyArchive").build()
+            then: "tar task executed and target folder contains plain tar"
+            result.task(':buildProducerTar').outcome == TaskOutcome.SUCCESS
+            result.task(':consumer:copyArchive').outcome == TaskOutcome.SUCCESS
+            file("producer-tar/build/distributions/opensearch-min.tar.gz").exists()
+            file("consumer/build/archives/opensearch-min.tar.gz").exists()
+
+            when:
+            result = gradleRunner("copyDir", "-Pversion=1.0").build()
+            then: "plain copy task executed and target folder contains plain content"
+            result.task(':buildProducer').outcome == TaskOutcome.SUCCESS
+            result.task(':consumer:copyDir').outcome == TaskOutcome.SUCCESS
+            file("producer-tar/build/install/someFile.txt").exists()
+            file("producer-tar/build/install/snapshot-1.0.txt").exists()
+            file("consumer/build/dir/someFile.txt").exists()
+
+            when:
+            gradleRunner("copyDir", "-Pversion=2.0").build()
+            then: "old content is cleared out"
+            file("producer-tar/build/install/someFile.txt").exists()
+            !file("producer-tar/build/install/snapshot-1.0.txt").exists()
+            file("producer-tar/build/install/snapshot-2.0.txt").exists()
+        } else {
+            when:
+            def result = gradleRunner("copyArchive").build()
```

Comment:
I think we should add a test case for Windows ZIP archive (for Windows platform), to check the archive distribution also works on Windows  (we had issues with that in the past), what do you think @Poojita-Raj?

Reply:
Hey @reta, thanks for the review! Yes, I think we can do that, however do you believe that's a blocker for windows readiness for the 1.3.7 release? Or would it be a nice to have add on?

Reply:
Thanks @Poojita-Raj , I think it is nice to have but certainly not a blocker

Reply:
Thanks @reta, I've opened another issue for your comment to track it, and will merge this in for now.